### PR TITLE
Keep lists of integers as lists when inspecting queries

### PR DESCRIPTION
**Pull request by michalmuskala**
@josevalim I'm opening this as a PR as I'm not sure about this change.

Is there a better way than to pattern match on the result of `Macro.to_string/2`?

### CODE REVIEW

File: lib/ecto/query/inspect.ex [hunk 1]
```diff
@@ -155,7 +155,12 @@ defimpl Inspect, for: Ecto.Query do
         {value, _type} -> Macro.escape(value)
         _              -> {:..., [], nil}
       end
-    Macro.to_string {:^, [], [escaped]}
+    case Macro.to_string({:^, [], [escaped]}) do
```
Comment (josevalim) [inline lib/ecto/query/inspect.ex]: I think we should change the whole function to something like this:

```elixir
  defp expr_to_string({:^, _, [ix]}, _, _, %{params: params}) do
    case Enum.at(params || [], ix) do
      {value, _type} -> "^" <> inspect(value, char_lists: :as_lists)
      _              -> "^..."
    end
  end
```

Thoughts?

Comment (michalmuskala) [inline lib/ecto/query/inspect.ex]: Makes sense

Comment (michalmuskala) [inline lib/ecto/query/inspect.ex]: Should we `Macro.escape` the value before inspecting?

Comment (josevalim) [inline lib/ecto/query/inspect.ex]: I don't think so. Macro escaping is actually forcing it to be inspected.

**Comment by michalmuskala** [comment]
@josevalim updated

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**